lib/az-bunyan/rotation.ts

```diff
@@ -0,0 +1,107 @@
+import { TableBatch, TableService, TableUtilities, TableQuery } from 'azure-storage';
+import * as moment from 'moment';
+
+export interface IAzurTableStorageLogRotation {
+	daysToKeep: number;
+	cleanupIntervalHours: number;
+}
+
+export interface ITableStorageLogRotationOptions {
+	cleanupIntervalHours: number;
+	daysToKeep: number;
+	tableName: string;
+	tableService: TableService;
+}
+
+export class TableStorageLogRotation {
+	private cleanupIntervalHours: number;
+	private daysToKeep: number;
+	private tableName: string;
+	private tableService: TableService;
+
+	constructor(options: ITableStorageLogRotationOptions) {
+		this.cleanupIntervalHours = options.cleanupIntervalHours;
+		this.daysToKeep = options.daysToKeep;
+		this.tableName = options.tableName;
+		this.tableService = options.tableService;
+
+		this.bootstrap();
```

Comment:
I find that constructor doing heavy lifting is not a great thing. I would give the responsibility to call bootstrap to the consumer.